src/python/zquantum/core/wip/circuit/conversions/qiskit_conversions.py

```diff
@@ -52,11 +55,35 @@
     XX: qiskit.extensions.RXXGate,
     YY: qiskit.extensions.RYYGate,
     ZZ: qiskit.extensions.RZZGate,
+    # TODO: add CustomGate mapping after we decide on "Gate|GateOperation" split
 }
 
 
+def _make_controlled_gate_factory(gate_cls):
+    def _factory(control, *gate_args):
+        return ControlledGate(gate_cls(*gate_args), control)
+
+    return _factory
+
+
+# NOTE:
+# Qiskit implements a special subclasses for some of the controlled gate variants:
+# https://github.com/Qiskit/qiskit-terra/blob/0.16.4/qiskit/circuit/add_control.py#L72
+# We'll support most of them eventually. The development plan is:
+# 1. Add mapping for some most common cases (done).
+# 2. Support most of the Qiskit controlled gates via our CustomGate when it's reworked.
+# 3. Add specialized gate subclasses on our side if there's a need for handling special
+#     cases explicitly.
+
+
 QISKIT_TO_ORQUESTRA_MAPPING = {
```

Comment:
I think we don't use "ORQUESTRA" anywhere in the code really, rather "zquantum".
TBH I'm not sure whether that's a good or bad decision, just pointing this out.
CC: @max-radin

Reply:
The `ORQUESTRA` word here follows a convention used previously in other reworked conversion modules (`cirq_conversions`, `pyquil_conversions`), as well as some other functions and docstrings under the `wip` package. TBH I like `zquantum` better instead of `orquestra` in the code here because it would be consistent with both the repo and the package name. If we agree about it, I'd do this in another PR. WDYT, @dexter2206?

Reply:
I definitely won't mind that, but still would like to learn from @max-radin whether there are some other good reasons for not including the keyword "orquestra" in our python codebase :)

Reply:
At some point in the past we decided to drop `orquestra` from our python libraries so that people wouldn't mix up libraries that can be run on the platform with the underlying platform itself. Maybe someday we'll rename things, but for now I think it would be good to use `zquantum` here for consistency.

Reply:
I've done the rename for functions and docstrings under the `wip` package (38c22f4). It was quick and we don't risk much by doing it already:
- `ORQUESTRA` -> `ZQUANTUM`
- `Orquestra` -> `ZQuantum`
- `orquestra` -> `zquantum`